geode-core/src/main/java/org/apache/geode/internal/cache/wan/GatewaySenderAttributes.java

```diff
@@ -31,63 +31,201 @@
 
-  public ArrayList<GatewayTransportFilter> transFilters = new ArrayList<GatewayTransportFilter>();
+  private ArrayList<GatewayTransportFilter> transFilters = new ArrayList<GatewayTransportFilter>();
 
-  public List<AsyncEventListener> listeners = new ArrayList<AsyncEventListener>();
+  private List<AsyncEventListener> listeners = new ArrayList<AsyncEventListener>();
 
-  public GatewayEventSubstitutionFilter eventSubstitutionFilter;
+  private GatewayEventSubstitutionFilter eventSubstitutionFilter;
 
-  public String id;
+  private String id;
 
-  public int remoteDs = GatewaySender.DEFAULT_DISTRIBUTED_SYSTEM_ID;
+  private int remoteDs = GatewaySender.DEFAULT_DISTRIBUTED_SYSTEM_ID;
 
-  public LocatorDiscoveryCallback locatorDiscoveryCallback;
+  private LocatorDiscoveryCallback locatorDiscoveryCallback;
 
-  public boolean isDiskSynchronous = GatewaySender.DEFAULT_DISK_SYNCHRONOUS;
+  private boolean isDiskSynchronous = GatewaySender.DEFAULT_DISK_SYNCHRONOUS;
 
-  public OrderPolicy policy;
+  private OrderPolicy policy;
 
-  public int dispatcherThreads = GatewaySender.DEFAULT_DISPATCHER_THREADS;
+  private int dispatcherThreads = GatewaySender.DEFAULT_DISPATCHER_THREADS;
 
-  public int parallelism = GatewaySender.DEFAULT_PARALLELISM_REPLICATED_REGION;
+  private int parallelism = GatewaySender.DEFAULT_PARALLELISM_REPLICATED_REGION;
 
-  public boolean isParallel = GatewaySender.DEFAULT_IS_PARALLEL;
+  private boolean isParallel = GatewaySender.DEFAULT_IS_PARALLEL;
 
-  public boolean groupTransactionEvents = GatewaySender.DEFAULT_MUST_GROUP_TRANSACTION_EVENTS;
+  private boolean groupTransactionEvents = GatewaySender.DEFAULT_MUST_GROUP_TRANSACTION_EVENTS;
 
-  public boolean isForInternalUse = GatewaySender.DEFAULT_IS_FOR_INTERNAL_USE;
+  private boolean isForInternalUse = GatewaySender.DEFAULT_IS_FOR_INTERNAL_USE;
 
-  public boolean isBucketSorted = GatewaySenderAttributes.DEFAULT_IS_BUCKETSORTED;
+  private boolean isBucketSorted = GatewaySenderAttributes.DEFAULT_IS_BUCKETSORTED;
 
-  public boolean isMetaQueue = GatewaySenderAttributes.DEFAULT_IS_META_QUEUE;
+  private boolean isMetaQueue = GatewaySenderAttributes.DEFAULT_IS_META_QUEUE;
 
-  public boolean forwardExpirationDestroy = GatewaySender.DEFAULT_FORWARD_EXPIRATION_DESTROY;
+  private boolean forwardExpirationDestroy = GatewaySender.DEFAULT_FORWARD_EXPIRATION_DESTROY;
 
-  public boolean enforceThreadsConnectSameReceiver =
+  private boolean enforceThreadsConnectSameReceiver =
       GatewaySender.DEFAULT_ENFORCE_THREADS_CONNECT_SAME_RECEIVER;
 
+  // Added due to "alter gateway-sender" command
+
+  private boolean modifyAlertThreshold = false;
+
+  private boolean modifyBatchSize = false;
+
+  private boolean modifyBatchTimeInterval = false;
+
+  private boolean modifyGroupTransactionEvents = false;
+
+  private boolean modifyGatewayEventFilter = false;
+
+  private boolean modifyGatewayTransportFilter = false;
```

Comment:
What is the purpose of these members and the functions to get/set them? I have not found the use in the rest of the code.

Reply:
Thanks for comments. This is not needed, since it was part of previous solution. I will remove it.